### PR TITLE
Refactor AI agent with Behavior tree

### PR DESCRIPTION
**Pull request by yuokamoto**
image (10)
image (9)
image (8)
image (7)

### CODE REVIEW

File: Source/RapyutaSimulationPlugins/Private/Robots/RRAIRobotROSController.cpp [hunk 1]
```diff
@@ -1095,13 +1100,48 @@ FTransform ARRAIRobotROSController::GetOrigin()
...
+}
+
 void ARRAIRobotROSController::ModeUpdate()
 {
     if (Mode != ERRAIRobotMode::MANUAL)
     {
         if (NavStatus == ERRAIRobotNavStatus::IDLE)
         {
             FMoveCompleteCallback onSuccess, onFail;    // dummy
+            FTransform dummy;
```
Comment (tetorea) [inline Source/RapyutaSimulationPlugins/Private/Robots/RRAIRobotROSController.cpp:1144]: It seems dummy is initialized after but never used

Comment (yuokamoto) [reply Source/RapyutaSimulationPlugins/Private/Robots/RRAIRobotROSController.cpp:1144]: it is used  at line 1155 and 1158

Comment (tetorea) [reply Source/RapyutaSimulationPlugins/Private/Robots/RRAIRobotROSController.cpp:1144]: these lines initialize dummy but then it is not used?

Comment (yuokamoto) [reply Source/RapyutaSimulationPlugins/Private/Robots/RRAIRobotROSController.cpp:1144]: Those are mainly used by Blueprint to get output transform directly.

**Review by tetorea — APPROVED** [review]

**Pull request merged**